velox/dwio/parquet/tests/reader/E2EFilterTest.cpp

```diff
@@ -85,6 +87,56 @@ class E2EFilterTest : public E2EFilterTestBase {
     return std::make_unique<ParquetReader>(std::move(input), opts);
   }
 
+  void testLocalFileSinkCloseAbort(bool isAbort) {
+    auto rowType = ROW({INTEGER()});
+    std::vector<RowVectorPtr> batches;
+    batches.push_back(std::static_pointer_cast<RowVector>(
+        test::BatchMaker::createBatch(rowType, 2, *leafPool_, nullptr, 0)));
+    LocalFileSink::registerFactory();
+    auto root = exec::test::TempDirectoryPath::create();
+    auto filePath = fs::path(root->path) / "xxx/yyy/zzz/test_file.ext";
+
+    ASSERT_FALSE(fs::exists(filePath.string()));
+    auto sink = FileSink::create(
+        fmt::format("file:{}", filePath.string()), {.pool = leafPool_.get()});
+    ASSERT_TRUE(sink->isBuffered());
+    EXPECT_TRUE(fs::exists(filePath.string()));
+
+    options_.memoryPool = rootPool_.get();
+    options_.flushPolicyFactory = [&]() {
+      return std::make_unique<LambdaFlushPolicy>(
+          rowsInRowGroup_, bytesInRowGroup_, [&]() { return false; });
+    };
+
+    auto sinkPtr = sink.get();
+    EXPECT_FALSE(sinkPtr->isClosed());
+
+    auto writer = std::make_unique<facebook::velox::parquet::Writer>(
+        std::move(sink), options_);
+    for (auto& batch : batches) {
+      writer->write(batch);
+    }
+    writer->flush();
+    auto size = sinkPtr->size();
+    EXPECT_EQ(size, fs::file_size(filePath));
+
+    batches.clear();
```

Comment:
why do we need to clear and create a new batch? Can't you create a single batch and write that twice?

Reply:
Fixed